### PR TITLE
Makes creating anonymous servers part of the public API

### PR DESCRIPTION
**Pull request by jakaplan**
Almost all of these changes are documentation related aside from changing the access levels from `internal` to `public`

### CODE REVIEW

**Comment by jakaplan** [comment]
@amomchilov In terms of reviewing, let me know if you think there's anything remaining that needs to happen before making anonymous servers public. Your feedback on the documentation is also welcome, but completely fine if you don't review that.

File: Sources/SecureXPC/XPCServerEndpoint.swift [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 //
 
 import Foundation
 
+/// An endpoint is used to create clients which can communicate with their associated server.
+///
+/// Endpoints are retrieved from a server's ``NonBlockingServer/endpoint`` property. They can be used in the same process or sent across an existing XPC
+/// connection.
+///
+/// > Warning: While ``XPCServerEndpoint`` conforms to `Codable` it can only be encoded and decoded by the `SecureXPC` framework.
```
Comment (amomchilov) [inline Sources/SecureXPC/XPCServerEndpoint.swift:15]: The Swift doc format has a style exactly for this:

```suggestion
/// - Warning:
```

Perhaps `Important` might be more suitable, IDK. You can see them all here: https://nshipster.com/swift-documentation/#additional-fields

File: Sources/SecureXPC/Client/XPCClient.swift [hunk 2]
```diff
@@ -13,8 +13,9 @@ import Foundation
 /// There are two different types of services you can communicate with using this client: XPC Services and XPC Mach services. If you are uncertain which
 /// type of service you're using, it's likely an XPC Service.
 ///
-/// **XPC Services**
+/// Clients can also be created from an ``XPCServerEndpoint`` which is the only way to create a client for an anonymous server.
```
Comment (amomchilov) [inline Sources/SecureXPC/Client/XPCClient.swift:16]:
```suggestion
/// Clients can also be created from an ``XPCServerEndpoint``, which is the only way to create a client for an anonymous server.
```

File: Sources/SecureXPC/Server/XPCServer.swift [hunk 3]
```diff
@@ -13,6 +13,8 @@ import Foundation
 /// There are two different types of services you can retrieve a server for: XPC Services and XPC Mach services. If you're uncertain which type of service you're
 /// using, it's likely an XPC Service.
 ///
+/// Anonymous servers can also be created which do not correspond to a service.
```
Comment (amomchilov) [inline Sources/SecureXPC/Server/XPCServer.swift:16]: This is a bit jargony IMO. I'm not sure "a server without a service" would make sense unless people were already in-the-know.

Perhaps:

```suggestion
/// Anonymous servers can also be created, which do not correspond to a Mach or XPC service. These can be useful for testing, or to aid in development by removing the need to debug across multiple processes.
```

Comment (amomchilov) [reply Sources/SecureXPC/Server/XPCServer.swift:16]: NVM, you cover use cases below, but you get the idea

File: Sources/SecureXPC/Server/XPCServer.swift [hunk 4]
```diff
@@ -312,7 +322,9 @@ public class XPCServer {
 public protocol NonBlockingServer {
     /// Begins processing requests received by this XPC server.
     func start()
-    /// Use an endpoint to create connections to this server
+    /// Use an endpoint to create connections to this server.
```
Comment (amomchilov) [inline Sources/SecureXPC/Server/XPCServer.swift:325]: This verb-phrase doesn't really cover what this property does itself (it talks about what you can do down the line). How about splitting that up, such as:

```suggestion
    /// Export an endpoint for this server, which can later be used to create new connections to this server.
```

File: Sources/SecureXPC/Server/XPCServer.swift [hunk 5]
```diff
@@ -341,12 +353,56 @@ extension XPCServer {
...
+    /// Creates a new anonymous server that accepts requests from the same process it's running in.
+    ///
+    /// Only a client created from an anonymous server's endpoint can communicate with that server. Retrieve the ``NonBlockingServer/endpoint`` and
+    /// then create a client with it:
+    /// ```swift
+    /// let server = XPCServer.makeAnonymous()
+    /// let client = XPCClient.fromEndpoint(server.endpoint)
+    /// ```
+    ///
+    /// > Important: No requests will be processed until ``NonBlockingServer/start()`` or ``startAndBlock()`` is called.
```
Comment (amomchilov) [inline Sources/SecureXPC/Server/XPCServer.swift:365]:
```
    /// - Important:
```

File: Sources/SecureXPC/Server/XPCServer.swift [hunk 6]
```diff
@@ -341,12 +353,56 @@ extension XPCServer {
...
+    ///
+    /// ## Usage
+    /// If you need this server to be communicated with by clients running in a different process, use ``makeAnonymous(clientRequirements:)`` instead.
```
Comment (amomchilov) [inline Sources/SecureXPC/Server/XPCServer.swift:368]: This isn't really "usage" information, it's kind of the opposite. Perhaps this should be an `Important`, `Attention`, `Note` or `Remark`?

File: Sources/SecureXPC/Server/XPCServer.swift [hunk 7]
```diff
@@ -341,12 +353,56 @@ extension XPCServer {
...
+    public static func makeAnonymous() -> XPCServer & NonBlockingServer {
         XPCAnonymousServer(messageAcceptor: SameProcessMessageAcceptor())
     }
 
-    internal static func makeAnonymous(clientRequirements: [SecRequirement]) -> XPCServer & NonBlockingServer {
+    /// Creates a new anonymous server that accepts requests from clients which meet the security requirements.
+    ///
+    /// Only a client created from an anonymous server's endpoint can communicate with that server. Retrieve the ``NonBlockingServer/endpoint`` and
```
Comment (amomchilov) [inline Sources/SecureXPC/Server/XPCServer.swift:375]: This is a bit of a sentence fragment, and doesn't fit in:


> "Retrieve the ``NonBlockingServer/endpoint`` and send it across an existing XPC connection."

File: Sources/SecureXPC/Server/XPCServer.swift [hunk 8]
```diff
@@ -341,12 +353,56 @@ extension XPCServer {
...
+    /// ```swift
+    /// let reqString = "identifier \"com.example.AuthorizedClient\" and certificate leaf[subject.OU] = \"4L0ZG128MM\""
+    /// var requirement: SecRequirement?
+    /// if SecRequirementCreateWithString(reqString as CFString,
+    ///                                   SecCSFlags(),
+    ///                                   &requirement) == errSecSuccess,
+    ///   let requirement = requirement {
+    ///     let server = XPCServer.makeAnonymous(clientRequirements: [requirement])
+    ///
+    ///    <# configure and start server #>
```
Comment (amomchilov) [inline Sources/SecureXPC/Server/XPCServer.swift:389]: Indentation:

```suggestion
    ///     let requirement = requirement {
    ///     let server = XPCServer.makeAnonymous(clientRequirements: [requirement])
    ///
    ///     <# configure and start server #>
```

**Review by amomchilov — APPROVED** [review]
Some, comments, though it's mainly subjective grammar stuff

File: Sources/SecureXPC/XPCServerEndpoint.swift (hunk 1 above)
Comment (jakaplan) [reply Sources/SecureXPC/XPCServerEndpoint.swift:15]: This is the [DocC format](https://developer.apple.com/documentation/xcode/formatting-your-documentation-content):
image

File: Sources/SecureXPC/Client/XPCClient.swift (hunk 2 above)
Comment (jakaplan) [reply Sources/SecureXPC/Client/XPCClient.swift:16]: I believe a comma in this case wouldn't be appropriate as the clause which follows is essential information. See https://www.grammarly.com/blog/comma-before-which/.

File: Sources/SecureXPC/Server/XPCServer.swift (hunk 3 above)
Comment (jakaplan) [reply Sources/SecureXPC/Server/XPCServer.swift:16]: Good call, will incorporate the first portion of this.

File: Sources/SecureXPC/Server/XPCServer.swift (hunk 4 above)
Comment (jakaplan) [reply Sources/SecureXPC/Server/XPCServer.swift:325]: Indeed, incorporated your feedback.

File: Sources/SecureXPC/Server/XPCServer.swift (hunk 5 above)
Comment (jakaplan) [reply Sources/SecureXPC/Server/XPCServer.swift:365]: Same note about this being the DocC format.

File: Sources/SecureXPC/Server/XPCServer.swift (hunk 6 above)
Comment (jakaplan) [reply Sources/SecureXPC/Server/XPCServer.swift:368]: Hmm, yeah I was unsure of how to refer to this. I'll use a `Note`.

File: Sources/SecureXPC/Server/XPCServer.swift (hunk 7 above)
Comment (jakaplan) [reply Sources/SecureXPC/Server/XPCServer.swift:375]: Yeah it does read a bit awkwardly, I've changed it to:
> Do this by retrieving the server’s `NonBlockingServer/endpoint` and then creating a client with it:

File: Sources/SecureXPC/Server/XPCServer.swift (hunk 8 above)
Comment (jakaplan) [reply Sources/SecureXPC/Server/XPCServer.swift:389]: Good catch, fixed.

File: Sources/SecureXPC/XPCServerEndpoint.swift (hunk 1 above)
Comment (amomchilov) [reply Sources/SecureXPC/XPCServerEndpoint.swift:15]: TIL. Nice!

**Pull request merged**